### ISSUE TITLE
Missing lines in full join of multiple tables

### ISSUE THREAD

**Issue by Columbus240** [pola-rs/r-polars#1246]
There might be an inconsistency in the library. The following example joins three `LazyFrame`s together with a full-join. Missing values in one column are filled. If the resulting `LazyFrame` is written to a file via `sink_*` (I tested both csv and parquet), the result is different than first calling `collect()` and then `write_*`. Namely, when using `sink_*`, some rows are missing in the output. It looks to me, like the query optimizer falsely replaces the full-join with an inner-join.  When using `collect()$write_*`, the computation seems correct to me.

Is this a known problem? Please feel free to ask, if I shall provide more info.

```R
library(polars)

df_A = pl$DataFrame(
  K = c(TRUE, FALSE),
  A = c(1L, 1L)
)

df_B = pl$DataFrame(
  K = c(TRUE),
  B = c(1L)
)

df_C = pl$DataFrame(
  K = c(TRUE),
  C = c(1L)
)

df_ABC =
  df_A$lazy()$join(
    df_B$lazy(), how = 'full', on = c('K'), coalesce = TRUE
  )$join(
    df_C$lazy(), how = 'full', on = c('K'), coalesce = TRUE
  )$with_columns(
    B = pl$col('B')$fill_null(0L)
  )

file_abc = 'file_abc.csv'
file_abc_collect = 'file_abc_collect.csv'

df_ABC$sink_csv(file_abc)
df_ABC$collect()$write_csv(file_abc_collect)

df_read = pl$read_csv(file_abc)
df_read_c = pl$read_csv(file_abc_collect)

df_read$height # returns 1, but should be 2
df_read_c$height # returns 2
```

The output of `polars_info()`:
```
Polars R package version : 0.19.1
Rust Polars crate version: 0.42.0

Thread pool size: 14 

Features:                               
default                    TRUE
full_features              TRUE
disable_limit_max_threads  TRUE
nightly                    TRUE
sql                        TRUE
rpolars_debug_print       FALSE

Code completion: deactivated 
```

The problem does not seem to appear, when only two dataframes are joined. Removing the `fill_null` also makes the problem disappear.

**Comment by etiennebacher** [comment]
Thanks for the report and the nice example. I can reproduce this in R and in Python, so this issue probably comes from upstream. I'll report an issue in python polars.

**Comment by etiennebacher** [comment]
I still see the error without the `$fill_null()` call though

**Comment by etiennebacher** [comment]
https://github.com/pola-rs/polars/issues/19040

**Comment by etiennebacher** [comment]
This was fixed upstream so it will be fixed here next time we update rust-polars

**Comment by Columbus240** [comment]
It is my displeasure to inform you, that I can still reproduce the above problem, even with the new release of r-polars.
My current output of `polars_info()`:
```
Polars R package version : 0.20.0
Rust Polars crate version: 0.43.1

Thread pool size: 14 

Features:                               
default                    TRUE
full_features              TRUE
disable_limit_max_threads  TRUE
nightly                    TRUE
sql                        TRUE
rpolars_debug_print       FALSE

Code completion: deactivated 
```

**Comment by eitsupi** [comment]
pola-rs/polars#19046 is not included Rust release and R package yet.

**Comment by Columbus240** [comment]
Ah, thanks for pointing this out. I was not aware.
I see now, that 19046 is only included in _Python_ Polars 1.9.0, but not yet in Rust Polars. I'm sorry for mixing up the release infos. (https://github.com/pola-rs/polars/releases)

**Comment by eitsupi** [comment]
Actually we could take in any of the commits, we just haven't done so because it would be quite a bit of work.
If you would like to include the py-1.9.0 release, pull requests are welcome.

**Comment by etiennebacher** [comment]
This is fixed in `polars` 0.21.0
